File: src/main/kotlin/texlab/Language.kt
package texlab

enum class Language {
    LATEX,
    BIBTEX;
}

fun getLanguageById(id: String): Language? {
    return when (id) {
        "latex" ->
            Language.LATEX
        "bibtex" ->
            Language.BIBTEX
        else ->
            null
    }
}

fun getLanguageByExtension(extension: String): Language? {
    return when (extension.toLowerCase()) {
        "tex" ->
            Language.LATEX
        "sty" ->
            Language.LATEX
        "cls" ->
            Language.LATEX
        "bib" ->
            Language.BIBTEX
        else ->
            null
    }
}

File: crates/tex/src/language.rs
#[derive(Debug, PartialEq, Eq, Clone, Copy)]
pub enum Language {
    Latex,
    Bibtex,
}

impl Language {
    pub fn by_extension(extension: &str) -> Option<Self> {
        match extension.to_lowercase().as_ref() {
            "tex" | "sty" | "cls" | "lco" | "aux" => Some(Language::Latex),
            "bib" => Some(Language::Bibtex),
            _ => None,
        }
    }

    pub fn by_language_id(language_id: &str) -> Option<Self> {
        match language_id {
            "latex" | "tex" => Some(Language::Latex),
            "bibtex" | "bib" => Some(Language::Bibtex),
            _ => None,
        }
    }
}
